standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/client/builder/GetPartitionProjectionsSpecBuilder.java

```diff
@@ -41,6 +42,12 @@ public GetPartitionProjectionsSpecBuilder addProjectField(String field) {
         return this;
     }
 
+    public GetPartitionProjectionsSpecBuilder addProjectFieldList(List<String> fields) {
+        fieldList.addAll(Arrays.asList("catName","dbName","tableName"));
```

Comment:
why need these fields here?

Reply:
@wecharyu In case of sending an empty list, our API will fetch the whole object. Hence, we are using basic catName, dbName, tblName for cases like, we just need to check that partition exists.